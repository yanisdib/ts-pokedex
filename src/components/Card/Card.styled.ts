import styled from 'styled-components';


interface Props {
    cardColor?: string;
}


export const StyledCard = styled.div`
    display: flex;
    justify-content: center;
    width: fit-content;
    height: fit-content;
    min-height: 400px;
    border: ${({
        cardColor
    }: Props) => `solid 2px ${cardColor ?? '#FFFF00'}`};

    @media (min-width: 375px) {
        min-width: 200px;
        border-radius: 15px;
    }

    @media (min-width: 744px) {
        min-width: 220px;
        border-radius: 18px;
    }

    @media (min-width: 950px) {
        min-width: 280px;
        border-radius: 20px;
    }
`;

export const Wrapper = styled.div`
    display: flex;
    width: inherit;
    height: inherit;
`;

export const Content = styled.div`
    display: flex;
    row-gap: 20px;
    width: inherit;
    height: inherit;
`;