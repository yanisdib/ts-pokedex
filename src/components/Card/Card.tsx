import { Content, StyledCard, Wrapper } from './Card.styled';


interface Props {
    cardColor?: string;
    children?: React.ReactNode;
}


function Card({ cardColor, children }: Props): JSX.Element {
    return (
        <StyledCard cardColor={cardColor} data-testid="card-test">
            <Wrapper>
                <Content>
                    {children}
                </Content>
            </Wrapper>
        </StyledCard>
    );
}


export default Card;