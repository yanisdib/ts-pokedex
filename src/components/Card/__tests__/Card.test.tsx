import { render } from '@testing-library/react';

import Card from '../Card';


describe('Card component', () => {
    test('should render without crashing', () => {
        const { container } = render(<Card />);

        expect(container).toMatchSnapshot();
    });

    test('should render children text', () => {
        const { getByText } = render(<Card>Hello there!</Card>);

        expect(getByText('Hello there!')).toBeInTheDocument();
    });

    test('should have yellow borders by default', () => {
        const { getByTestId } = render(<Card />);

        expect(getByTestId('card-test')).toHaveStyle('border: 2px solid #FFFF00');
    });

    test('should have red borders', () => {
        const { getByTestId } = render(<Card cardColor="red" />);

        expect(getByTestId('card-test')).toHaveStyle('border: 2px solid red');
    });
});