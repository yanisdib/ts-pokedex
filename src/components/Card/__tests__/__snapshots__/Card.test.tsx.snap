// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render without crashing 1`] = `
<div>
  <div
    class="sc-bczRLJ dFyAAJ"
    data-testid="card-test"
  >
    <div
      class="sc-gsnTZi lbrWgB"
    >
      <div
        class="sc-dkzDqf efhOyy"
      />
    </div>
  </div>
</div>
`;
